spp_import_dci_api/models/fetch_crvs_beneficiary.py

```diff
@@ -0,0 +1,1015 @@
+                    self.update_or_create_location(identifier, crvs_location_vals)
+
+    @api.model
+    def create_locations(self):
+        """
+        The function creates a URL by concatenating the URL of a data source with a location path.
+        """
+        data_source_id = self.env["spp.data.source"].search(
+            [("name", "=", constants.DATA_SOURCE_NAME)], limit=1
+        )
+
+        if data_source_id:
+            location_path = ""
+            data_source_path_id = self.env["spp.data.source.path"].search(
+                [
+                    ("name", "=", constants.DATA_SOURCE_LOCATION_PATH_NAME),
+                    ("data_source_id", "=", data_source_id.id),
+                ],
+                limit=1,
+            )
+            if data_source_path_id:
+                location_path = data_source_path_id.path
+
+            if location_path:
+                url = f"{data_source_id.url}{location_path}"
+                response = requests.get(url, timeout=constants.REQUEST_TIMEOUT)
+                if response.ok:
+                    result = response.json()
+                    self.process_location(result)
+
+    def get_parent(self):
+        """
+        The function returns the parent object if the current object has child objects, otherwise it
+        returns the parent_id of the current object.
+        :return: The code is returning either the current object (`self`) if it already has child
+        records, or the parent object (`self.parent_id`) if it does not have any child records.
+        """
+        self.ensure_one()
+
+        if self.child_ids:
+            # return self if already a parent
+            return self
+        else:
+            # return parent if self doesn't have a child
+            return self.parent_id
+
+
+class SPPFetchCRVSBeneficiary(models.Model):
+    _name = "spp.fetch.crvs.beneficiary"
+    _description = "Fetch CRVS Beneficiary"
+
+    def _get_default_domain(self):
+        return (
+            f'["&",["birthdate",">","{constants.DEFAULT_YEAR}-{constants.DEFAULT_MONTH}-{constants.DEFAULT_DAY}"]'
+            f',["birthdate","<","{constants.DEFAULT_YEAR}-{constants.DEFAULT_MONTH}-{constants.DEFAULT_DAY}"]]'
+        )
+
+    data_source_id = fields.Many2one(
+        "spp.data.source",
+        required=True,
+        default=lambda self: self.env["spp.data.source"].search(
+            [("name", "=", constants.DATA_SOURCE_NAME)], limit=1
+        ),
+        readonly=True,
+    )
+
+    name = fields.Char("Search Criteria Name", required=True)
+
+    location_type_id = fields.Many2one("spp.crvs.location.type")
+    location_id = fields.Many2one("spp.crvs.location")
+    location_id_domain = fields.Char(
+        compute="_compute_location_id_domain",
+        readonly=True,
+    )
+
+    domain = fields.Text(
+        string="Filter",
+        default=_get_default_domain,
+        required=True,
+    )
+
+    done_imported = fields.Boolean()
```

Comment:
```suggestion
    done_imported = fields.Boolean(readonly=True)
```
I don't think this should be changed by end user